### PR TITLE
restructure  cmake workflow presets

### PR DESCRIPTION
**Pull request by ClausKlein**
Add codespell config file and fix some typos too

Prevent build errors with gcc on Darwin (macOS) CI

Cleanup the cmake files

Add the cmake workflow presets for Windows too.

### CODE REVIEW

**Comment by ClausKlein** [comment]
@dietmarkuehl Would you like presets like this https://github.com/friendlyanon/cmake-init-shared-static/blob/master/CMakePresets.json

I would prefer this one: https://github.com/ClausKlein/cpp-lib-template/blob/develop/CMakePresets.json

As toolchain for Windows CI build with ninja normally I use the [WindowsToolchain](https://github.com/MarkSchofield/WindowsToolchain) as git submodule or as copy?

**Comment by dietmarkuehl** [comment]
> @dietmarkuehl Would you like presets like this https://github.com/friendlyanon/cmake-init-shared-static/blob/master/CMakePresets.json
> 
> I would prefer this one: https://github.com/ClausKlein/cpp-lib-template/blob/develop/CMakePresets.json
> 
> As toolchain for Windows CI build with ninja normally I use the [WindowsToolchain](https://github.com/MarkSchofield/WindowsToolchain) as git submodule or as copy?

@ClausKlein I don't have an opinion on the presets! The ones I have were just copied from `beman.exemplar` when I created the repository. I don't think I normally build using the presets, either (it is my understanding your previous PRs changed that). At some point I should probably understand what they are and how they work.

In general, I think Beman projects should follow what `beman.exemplar` does. I like consistency. There may be some reasons to deviate. On the other hand, if things are done consistently it is easier to layer additional tool on top. For example, I know there is a desire to package things with Conan - another tool I haven't used and I don't know what it actually does.

**Comment by dietmarkuehl** [comment]
It seems the `delete(message)` is causing grief with LLVM 19: I wouldn’t object to disabling the conditional check using the feature test macro for now. I can restore that in a different PR.

File: CMakePresets.json [hunk 1]
```diff
@@ -1,80 +1,27 @@
...
-        "root-config",
-        "release-base-unix"
-      ]
-    }
+  "version": 9,
+  "cmakeMinimumRequired": {
+    "major": 3,
+    "minor": 30,
+    "patch": 0
+  },
```
Comment (dietmarkuehl) [inline CMakePresets.json:7]: These recent versions prevent me from building on Debian 12 which installs `cmake` version 3.25.1. I'd think others would have the same problem.

**Review by dietmarkuehl — APPROVED** [review]
First off: thank you very much for this work! There are number of things here I wanted to do for a while but didn't get to do, yet!

I didn't spot anything critical. There are a few comments but I'd be OK to land this fairly large PR and possibly address them later. The only one I consider really odd is the use of `sudo sudo` [without a user] although I don't think it causes any harm.

It seems the presets stuff needs fairly recent versions of `cmake`, certainly a more recent version than on the Debian 12 Linux I'm using (it has `cmake 3.25` installed). Using a non-presets build does work, though.

As there are some comments, I won't merge the PR: feel free to merge (or let me know if you can't or want me to merge).

**Review by dietmarkuehl — APPROVED** [review]
Looks good

**Pull request merged**